lib/foreman_rh_cloud.rb

```diff
@@ -3,17 +3,23 @@
 require 'uri'
 
 module ForemanRhCloud
+  def self.on_premise_url
+    return unless ForemanRhCloud.with_insights_on_premise?
+    port = ENV['INSIGHTS_ADVISOR_PORT'] || "8000"
+    "http://#{ForemanRhCloud.foreman_host.fqdn || 'localhost'}:#{port}"
+  end
+
   def self.base_url
     # for testing set ENV to 'https://ci.cloud.redhat.com'
-    @base_url ||= ENV['SATELLITE_RH_CLOUD_URL'] || 'https://cloud.redhat.com'
+    @base_url ||= on_premise_url || ENV['SATELLITE_RH_CLOUD_URL'] || 'https://cloud.redhat.com'
```

Comment:
```suggestion
    @base_url ||= ENV['SATELLITE_RH_CLOUD_URL'] || on_premise_url || 'https://cloud.redhat.com'
```

I think we should leave the option of explicitly defining the URL even if the port is defined

Reply:
updated :+1: